File: c++/Easy/Part005/Identify Correct Operator.cpp
#include <iostream>
#include <string>
#include <cstdlib>
using namespace std;

int main() {
    // Read the input expression
    string expression;
    cin >> expression;

    // Identify the position of the operator and the equal sign
    size_t operatorPos = expression.find_first_of("+-*/");
    size_t equalPos = expression.find('=');

    // Extract numbers as strings
    string aStr = expression.substr(0, operatorPos);
    string bStr = expression.substr(operatorPos + 1, equalPos - operatorPos - 1);
    string cStr = expression.substr(equalPos + 1);

    // Convert strings to integers
    int A = stoi(aStr);
    int B = stoi(bStr);
    int C = stoi(cStr);

    // Determine the correct operator
    if (A + B == C) {
        cout << "+" << endl;
    } else if (A - B == C) {
        cout << "-" << endl;
    } else if (A * B == C) {
        cout << "*" << endl;
    } else if (B != 0 && A / B == C) { // Check for division by zero
        cout << "/" << endl;
    } else {
        cout << "No valid operator found." << endl; // This shouldn't happen as per problem constraints
    }

    return 0;
}

File: Java/Easy/Part004/Identify Correct operator.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        // Read the input expression
        String expression = scanner.next();

        // Identify the position of the operator and the equal sign
        int operatorPos = -1;
        int equalPos = -1;

        // Find operator position
        for (int i = 0; i < expression.length(); i++) {
            char c = expression.charAt(i);
            if (c == '+' || c == '-' || c == '*' || c == '/') {
                operatorPos = i;
                break;
            }
        }

        // Find equal sign position
        for (int i = 0; i < expression.length(); i++) {
            if (expression.charAt(i) == '=') {
                equalPos = i;
                break;
            }
        }

        // Extract numbers as strings
        String aStr = expression.substring(0, operatorPos);
        String bStr = expression.substring(operatorPos + 1, equalPos);
        String cStr = expression.substring(equalPos + 1);

        // Convert strings to integers
        int A = Integer.parseInt(aStr);
        int B = Integer.parseInt(bStr);
        int C = Integer.parseInt(cStr);

        // Determine the correct operator
        if (A + B == C) {
            System.out.println("+");
        } else if (A - B == C) {
            System.out.println("-");
        } else if (A * B == C) {
            System.out.println("*");
        } else if (B != 0 && A / B == C) { // Check for division by zero
            System.out.println("/");
        } else {
            System.out.println("No valid operator found."); // This shouldn't happen as per problem constraints
        }

        scanner.close();
    }
}
